### PR TITLE
Adding UIKit extensions tests

### PR DESCRIPTION
**Pull request by FranDepascuali**
## Sumary ##
**Trello card**: https://trello.com/c/5ldD3wbp/10-agregar-tests

Adds UIKit's extensions tests.

## Known issues ##
This are unit tests and not everything could be tested.

### CODE REVIEW

**Comment by woloxci** [comment]
:+1: There are no linter violations.

File: CoreTests/Extensions/UIKit/UIAlertControllerSpec.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                }
+                
+                it("should return confirmationAlertViewModel's title") {
+                    expect(alertController.title).to(equal(confirmationAlertViewModel.title))
+                }
+                
+                it("should return confirmationAlertViewModel's message") {
+                    expect(alertController.message).to(equal(confirmationAlertViewModel.message))
+                }
+
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UIAlertControllerSpec.swift]:
```
it("should have two possible actions: cancel and confirm") {
    expect(alertController.actions.count).to(equal(2))
}
```

File: CoreTests/Extensions/UIKit/UIColorSpec.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                        let color = UIColor(hex: "#08BBF4")
+                        
+                        expect(color).toNot(beNil())
+                    }
+                }
+            }
+            
+            context("When the hexString is invalid") {
+                
+                context("When the length is greater than 6") {
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UIColorSpec.swift]: If I remember correctly the possible lengths are 6, or 8 if there is specification for the alpha component.

File: CoreTests/Extensions/UIKit/UITextFieldSpec.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                beforeEach {
+                    textField = UITextField()
+                    let nextTextField = UITextField()
+                    nextTextField.text = "next"
+                    textField.next = nextTextField
+                }
+                
+                it("should return the next textfield") {
+                    expect(textField.next).toNot(beNil())
+                    expect(textField.next?.text!).to(equal("next"))
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UITextFieldSpec.swift]: I think you could directly compare `expect(textField.next!).to(equal(nextTextField))` if you store it.

File: CoreTests/Extensions/UIKit/UITextFieldSpec.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                context("When the next textField is changed") {
+                    
+                    beforeEach {
+                        let nextTextField = UITextField()
+                        nextTextField.text = "changed"
+                        textField.next = nextTextField
+                    }
+                    
+                    it("should return the new textfield") {
+
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UITextFieldSpec.swift]: Remove empty lines.

File: CoreTests/Extensions/UIKit/UITextFieldSpec.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                    beforeEach {
+                        let nextTextField = UITextField()
+                        nextTextField.text = "changed"
+                        textField.next = nextTextField
+                    }
+                    
+                    it("should return the new textfield") {
+                        
+                        expect(textField.next).toNot(beNil())
+                        print(textField.next!.text)
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UITextFieldSpec.swift]: Remove print.

File: CoreTests/Extensions/UIKit/UITextFieldSpec.swift [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                        expect(textField.next?.text!).to(equal("changed"))
```
Comment (danielaRiesgo) [inline CoreTests/Extensions/UIKit/UITextFieldSpec.swift]: Same about direct comparison to textfield.

**Comment by danielaRiesgo** [comment]
👍

**Pull request merged**